EX1/HW6.py

```diff
@@ -0,0 +1,208 @@
+import time
+import os
+import numpy as np
+from PIL import Image
+
+
+def read_words_list(filename):
+    with open(filename, 'r') as file:
+        words = file.read().split()
+    return words
+
+
+def read_words_set(filename):
+    with open(filename, 'r') as file:
+        words = set(file.read().split())
+    return words
+
+
+def compare_read_words(filename):
+    list_start_time = time.time()
+    words_list = read_words_list(filename)
+    list_end_time = time.time()
+    list_elapsed_time = list_end_time - list_start_time
+
+    set_start_time = time.time()
+    words_set = read_words_set(filename)
+    set_end_time = time.time()
+    set_elapsed_time = set_end_time - set_start_time
+
+    if list_elapsed_time < set_elapsed_time:
+        print("List is faster")
+        print("list: ", list_elapsed_time)
+        print("set: ", set_elapsed_time)
+    elif list_elapsed_time > set_elapsed_time:
+        print("Set is faster")
+        print("list: ", list_elapsed_time)
+        print("set: ", set_elapsed_time)
+    else:
+        print("Both are equally fast")
+        print("list: ", list_elapsed_time)
+        print("set: ", set_elapsed_time)
+
+    return
+
+
+def find_states_same_row(filename):
+    with open(filename, 'r') as file:
+        states = file.read().split()
+
+    rows = ['qwertyuiop', 'asdfghjkl', 'zxcvbnm']
+    same_row_states = []
+
+    for state in states:
+        row_found = False
+        for row in rows:
```

Comment:
You should aim to write everything in a single line, even if you can. 
This link https://towardsdatascience.com/you-should-stop-craving-for-python-one-liners-5dbf8e0de23d explains it well, even though I disagree with how important memory and runtime are in most python programs.